include/s3select.h

```diff
@@ -1100,6 +1118,44 @@ void push_data_type::operator()(s3select* self, const char* a, const char* b) co
   }
 }
 
+void push_substr_from::operator()(s3select* self, const char* a, const char* b) const
+{
+  std::string token(a, b);
+
+  __function* func = S3SELECT_NEW(self, __function, "substring_from", self->getS3F());
+
+  base_statement* expr = self->getAction()->exprQ.back();
+  self->getAction()->exprQ.pop_back();
+  func->push_argument(expr);
+
+  base_statement* inp_expr = self->getAction()->exprQ.back();
+  self->getAction()->exprQ.pop_back();
+  func->push_argument(inp_expr);
+
+  self->getAction()->exprQ.push_back(func);
+}  
+
+void push_substr_from_for::operator()(s3select* self, const char* a, const char* b) const
+{
+  std::string token(a, b);
+
+  __function* func = S3SELECT_NEW(self, __function, "substring_from", self->getS3F());
+
+  base_statement* expr = self->getAction()->exprQ.back();
+  self->getAction()->exprQ.pop_back();
+  func->push_argument(expr);
+
+  base_statement* inp_expr = self->getAction()->exprQ.back();
+  self->getAction()->exprQ.pop_back();
+  func->push_argument(inp_expr);
+
+  base_statement* inp_exp = self->getAction()->exprQ.back();
```

Comment:
typo:
use better name for variables  , it makes the code more readable 
in this case, it will describe the order of parameters and their roles.